rust/agama-lib/src/storage/client.rs

```diff
@@ -140,4 +155,57 @@ impl<'a> StorageClient<'a> {
 
         Ok(self.calculator_proxy.calculate(dbus_settings).await?)
     }
+
+    async fn build_device(
+        &self,
+        object: &(
+            OwnedObjectPath,
+            HashMap<OwnedInterfaceName, HashMap<std::string::String, OwnedValue>>,
+        ),
+    ) -> Result<Device, ServiceError> {
+        Ok(Device {
+            device_info: self.build_device_info(&object.0).await?,
+            component: None,
```

Comment:
Perhaps we could implement `Default` or, if not possible, have some kind of constructor.

Reply:
I am thinking that in the end I will add to `Device` struct method `try_from` that will construct itself from that Object Manager struct. and here will be just `Ok(object.into())`